sdk/core/core/internal/az_precondition_internal.h

```diff
@@ -68,9 +68,15 @@ AZ_NODISCARD AZ_INLINE bool az_span_is_valid(az_span span, int32_t min_size, boo
     return false;
   }
 
-  uint8_t* ptr = az_span_ptr(span);
+  uint8_t* const ptr = az_span_ptr(span);
   int32_t const span_size = az_span_size(span);
 
+  // Can't wrap over the end of the address space
+  if (ptr + span_size < ptr)
```

Comment:
Can you please verify that this doesn't get optimized out and change the behavior? It's possible this may not achieve our intended goal.

Also, can you explain, maybe with a code snippet/test case what edge case this protects us against?

I read that some compilers in C consider `(X + 1 < X)` as a given by definition and hence optimize this out, leading to undefined behavior.
See http://blog.llvm.org/2011/05/what-every-c-programmer-should-know.html
https://blog.regehr.org/archives/213

Reply:
Ok, now I rewrote it, it won't wrap an int.

Reply:
It protects us from the case (assuming that you have 16-bit address space)
```
(az_span) {
    .ptr = (uint8_t*)0xFFF0,
    .size = 100,
}
```

Reply:
Fixed, clarified.